src/LinuxTracing/TracerThread.cpp

```diff
@@ -56,6 +56,9 @@ TracerThread::TracerThread(const CaptureOptions& capture_options)
       trace_thread_state_{capture_options.trace_thread_state()},
       trace_gpu_driver_{capture_options.trace_gpu_driver()} {
   if (unwinding_method_ != CaptureOptions::kUndefined) {
+    uint32_t stack_dump_size = capture_options.stack_dump_size();
+    CHECK(stack_dump_size <= kMaxStackSampleUserSize && stack_dump_size > 0);
```

Comment:
Yeah, as I said in the previous comment, I don't know if like that a client can make the service crash by sending a wrong value. We could just reassign `stack_dump_size` and print an `ERROR` instead. I see there's already a `FAIL_IF` below so that can actually already happen, but I don't know if I like that either honestly. What do you think?

Reply:
Good point. Let's do what you suggested. Report an error and re-assign to the default.